blockchain-source/src/cardano/event.rs

```diff
@@ -157,3 +145,51 @@ impl<Tip> GetNextFrom for CardanoNetworkEvent<BlockEvent, Tip> {
         }
     }
 }
+
+impl BlockEvent {
+    pub(crate) fn from_serialized_block(raw_block: SerializedBlock) -> anyhow::Result<Self> {
+        let block = raw_block
+            .unserialize()
+            .context("failed to deserialize block");
+
+        if let Ok(block) = block {
+            let id = BlockId::new(block.header().hash().to_string());
+            let block_number = BlockNumber::new(block.header().block_number());
+
+            let parent_id = get_parent_id(&block.header());
+
+            Ok(BlockEvent {
+                raw_block,
+                id,
+                parent_id,
+                block_number,
+                slot_number: SlotNumber::new(block.header().slot()),
+                is_boundary_block: false,
+                // this is not in the header, and computing it requires knowing the network
+                // details, which makes implementing `Serialize` and `Deserialize`more complicated,
+                // unless we serialize this field too.
+                // it can be computed later inside carp, since we don't need this in the bridge.
```

Comment:
@gostkin this is what the `Era` is used for currently. Not really sure if doing it that way was better or not. Maybe it's better to compute the field inside the source, and then serialize it.